Automattic/newspack-plugin
# feat: new onboarding menu UI (v3)

### All Submissions:






### Changes proposed in this Pull Request:



Implements new onboarding UI design as mocked up by @thomasguillot.

Currently, this PR only implements the design changes in step 0 (Welcome screen) and the redesigned pagination UI. Pending a decision on the new onboarding steps, I've kept the onboarding steps the same as they currently are, except for renaming the titles and routes in the cases where the new steps have exact analogues in the existing steps ("About your publication" becomes "Settings", "Site Design" becomes "Design", etc.).

Mobile (I've interpreted the design mockup as a dropdown, but please let me know if this is incorrect):
<img width="376" alt="Screen Shot 2020-12-14 at 11 53 55 AM" src="https://user-images.githubusercontent.com/2230142/102122734-2ca88e00-3e03-11eb-851d-a89876cc1d3f.png">

Desktop:
<img width="1657" alt="Screen Shot 2020-12-14 at 11 54 17 AM" src="https://user-images.githubusercontent.com/2230142/102122753-316d4200-3e03-11eb-8cf3-4173dff8dba3.png">

Closes #697.

### How to test the changes in this Pull Request:

1. Install the plugin on a new WP site: checkout this branch and run `npm run release:archive` to build a ZIP file that can be installed on a new test site.
2. Go through the onboarding steps. The Welcome screen should display a Newspack logo at the top, and there should no longer be a "Not right now" secondary button here.
4. Proceed through all onboarding steps and confirm that the correct step is highlighted in the new UI in both desktop and mobile viewports.
5. The onboarding process should be the same as on `master`, with the exception that "About your publication" is now called "Settings", "Installation..." is now called "Integrations" and "Site Design" is now called "Design" per the new design mockups.
6. Since we've changed the structure of some route data: Test that if trying to navigate to an invalid onboarding route (e.g. change the hash at the end of the URL to some gibberish), you get redirected back to the Welcome screen.

### Other information:
